src/other/index.jl

```diff
@@ -145,8 +145,19 @@ end
 @inline Base.getindex(x::AbstractIndex, idx::AbstractRange{<:Integer}) = collect(Int, idx)
 @inline Base.getindex(x::AbstractIndex, ::Colon) = Base.OneTo(length(x))
 
-@inline Base.getindex(x::Index, idx::Symbol) = x.lookup[idx]
-@inline Base.getindex(x::Index, idx::AbstractVector{Symbol}) = [x.lookup[i] for i in idx]
+@inline function lookupname(l::Dict{Symbol, Int}, idx::Symbol)
+    i = get(l, idx, nothing)
+    if i === nothing
+        throw(BoundsError("column name $idx not found in the data frame"))
```

Comment:
Maybe add `:`? Not sure what convention we use in this kind of situation.

Also, not to be done now, but I've been thinking for a long time that it would be great to suggest appropriate column names if we detect a case mismatch, or even better if we can identify a similar name based on some distance metric. :-)

Reply:
Added `:` as indeed this is what we use it when printing errors (but note that we have just dropped `:` when showing `GroupedDataFrame` per your suggestion 😄).

I have implemented fuzzy matching listing columns that differ in at most one position in edit distance and disregarding case.